### ISSUE TITLE
Rar5: IArchiveEntryExtensions WriteTo does not write to stream in 0.33 (RarArchive)

### ISSUE THREAD

**Issue by olegfeferman** [adamhathcock/sharpcompress#755]
Hello,

I am trying to write a piece of code that verifies if user entered a correct password in case archive was password protected like this:

            using (var archive = RarArchive.Open(archiveFile, new ReaderOptions() { Password = pass }))
            {
                var entry = archive.Entries.FirstOrDefault(e => !e.IsDirectory);
                
                using (var stream = new MemoryStream( (int) entry.Size))
                {                    
                    entry.WriteTo(stream);                                        
                    if (stream.Length == 0)
                    {
                        ShowMessage("You have entered incorrect password.");
                    }
                }
            }

Problem is stream is always zero. Either it is a bug or I am doing something wrong.

**Comment by olegfeferman** [comment]
It reports correct size in bytes of that entry

**Comment by Erior** [comment]
I don't think you actually know if the password is correct until you try to decompress, that would give you a size, however it would be the wrong size / CRC, comparing the result with entry.Size would make more sense.
Btw, when I try this with encrypted header info, it throw an exception, you might need to catch that if you have such files.

**Comment by olegfeferman** [comment]
Thank you for the feedback. When I test with Rar.encrypted_filesAndHeader.rar or Rar.encrypted_filesOnly.rar it works OK (in first case I get exception in second stream.Length is greater than 0). 
What I did not realize before you sent your message is that I am only having problem with Rar5.encrypted_filesAndHeader.rar and Rar5.encrypted_filesOnly.rar. Actually before I was just testing with a test archive that I created with latest version of Winrar (6.23).
